File: Dialogs/DialogModelViewDemo/app/src/main/java/edu/cs4730/dialogmodelviewdemo/DataViewModel.java
package edu.cs4730.dialogmodelviewdemo;

import android.app.Application;
import android.util.Log;

import androidx.annotation.NonNull;
import androidx.lifecycle.AndroidViewModel;
import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;

public class DataViewModel extends ViewModel {
    private MutableLiveData<String> item1;
    private MutableLiveData<String> item2;
    private MutableLiveData<Boolean> yesno;

    public DataViewModel() {
        item1 = new MutableLiveData<String>();
        item2 = new MutableLiveData<String>();
        yesno = new MutableLiveData<Boolean>();
        item1.setValue("Nothing");
        item2.setValue("Nothing");
        yesno.setValue(false);
    }
    //first item
    LiveData<String> getItem1LD() {
        item1.setValue("Nothing");
        return item1;
    }

    String getItem1() {
        return item1.getValue();
    }

    void setItem1(String n) {
        Log.d("VM", "item1 is " + n);
        item1.setValue(n);
    }

    //second item
    LiveData<String> getItem2LD() {
        item2.setValue("Nothing");
        return item2;
    }

    String getItem2() {
        return item2.getValue();
    }

    void setItem2(String n) {
        Log.d("VM", "item2 is " + n);
        item2.setValue(n);
    }

    //and yesno boolean
    LiveData<Boolean> getYesNoLD() {
        yesno.setValue(false);
        return yesno;
    }

    Boolean getYesNo() {
        return yesno.getValue();
    }

    void setYesNo(Boolean n) {
        Log.d("VM", "YesNo is " + n);
        yesno.setValue(n);
    }

}

File: Dialogs/DialogViewModelDemo_kt/app/src/main/java/edu/cs4730/dialogviewmodeldemo_kt/DataViewModel.kt
package edu.cs4730.dialogviewmodeldemo_kt

import android.util.Log
import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel

class DataViewModel : ViewModel() {
    private val item1: MutableLiveData<String> = MutableLiveData("Nothing")
    private val item2: MutableLiveData<String> = MutableLiveData("Nothing")
    private val yesno: MutableLiveData<Boolean> = MutableLiveData(false)

    //first item
    val getItem1LD: LiveData<String>
        get() {
            item1.value = "Nothing"
            return item1
        }

    fun getItem1(): String? {
        return item1.value
    }

    fun setItem1(n: String) {
        Log.d("VM", "item1 is $n")
        item1.value = n
    }

    //second item
    val getItem2LD: LiveData<String>
        get() {
            item2.value = "Nothing"
            return item2
        }

    fun getItem2(): String? {
        return item2.value
    }

    fun setItem2(n: String) {
        Log.d("VM", "item2 is $n")
        item2.value = n
    }

    //and yesno boolean
    val getYesNoLD: LiveData<Boolean>
        get() {
            yesno.value = false
            return yesno
        }

    val yesNo: Boolean?
        get() = yesno.value

    fun setYesNo(n: Boolean) {
        Log.d("VM", "YesNo is $n")
        yesno.value = n
    }

}
